blacklist_service/src/test/kotlin/ru/zveron/service/BlacklistCrudServiceTest.kt

```diff
@@ -0,0 +1,169 @@
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import ru.zveron.AddRequest
+import ru.zveron.BlacklistProfile
+import ru.zveron.BlacklistTest
+import ru.zveron.DeleteAllRecordsWhereUserBlocksRequest
+import ru.zveron.DeleteAllRecordsWhereUserIsBlockedRequest
+import ru.zveron.DeleteRequest
+import ru.zveron.ExistInBlacklistRequest
+import ru.zveron.GetListRequest
+import ru.zveron.entity.BlacklistRecord
+import ru.zveron.exception.BlacklistException
+import ru.zveron.repository.BlacklistRepository
+
+class BlacklistCrudServiceTest : BlacklistTest() {
+
+    @Autowired
+    lateinit var blacklistRepository: BlacklistRepository
+
+    @Autowired
+    lateinit var blacklistCrudService: BlacklistCrudService
+
+    private fun launchBlocking(coroutine: suspend () -> Unit) = runBlocking { launch { coroutine() } }
+
+    @Test
+    fun `When check if record exists in blacklist and it exists return true`() {
+        launchBlocking {
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(1, 2)))
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(1, 3)))
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(2, 3)))
+
+            val result = blacklistCrudService.existInBlacklist(ExistInBlacklistRequest.newBuilder().setOwnerId(1).setTargetUserId(3).build())
+
+            result.exists shouldBe true
+        }
+    }
+
+    @Test
+    fun `When check if record exists in blacklist and it is not return false`() {
+        launchBlocking {
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(1, 2)))
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(1, 3)))
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(2, 3)))
+
+            val result = blacklistCrudService.existInBlacklist(ExistInBlacklistRequest.newBuilder().setOwnerId(1).setTargetUserId(4).build())
+
+            result.exists shouldBe false
+        }
+    }
+
+    @Test
+    fun `When get list of records got records`() {
+        launchBlocking {
+            val record1 = blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(1, 2)))
+            val record2 = blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(1, 3)))
+            blacklistRepository.save(BlacklistRecord(BlacklistRecord.BlacklistKey(2, 3)))
+
+            val set = blacklistCrudService.getList(GetListRequest.newBuilder().setId(1).build()).blacklistRecordsList.toSet()
+
+            set.size shouldBe 2
+            set.contains(BlacklistProfile.newBuilder().setId(record1.id.reportedId).build()) shouldBe true
+            set.contains(BlacklistProfile.newBuilder().setId(record2.id.reportedId).build()) shouldBe true
```

Comment:
set.map { it.id }.shouldContainExactlyInAnyOrder(record1.id.reportedId, record2.id.reportedId)